blocks/bricks/recurrent.py

```diff
@@ -630,3 +631,231 @@ def apply(self, *args, **kwargs):
     @apply.delegate
     def apply_delegate(self):
         return self.children[0].apply
+
+
+class RecurrentStack(BaseRecurrent, Initializable):
+    u"""Stack of recurrent networks.
+
+    Build a stack of recurrent layers from a supplied list of
+    BaseRecurrent objects. Each object must have a `sequences`,
+    `contexts`, `states` and `outputs` parameters to its `apply` method.
+    It is assumed that all `states` have a "states" element
+    (this can be configured with `states_name` parameter.)
+
+    The `sequences` (the "inputs") of the the first layer (layer 0) is
+    used as the `sequence` of the stack (the instance of this class.)
+    The "mask" element of the `sequences` is carried to all layers that
+    have a "mask" element in their `sequences`.
+    For all the following layers, all elements of the `sequences`, except
+    for "mask", are forked from the "states" element of the of the
+    `states` of the previous layer.
+
+    The concatenated `states` of all layers is the `states` of the stack.
+    In order to avoid conflict, the names of the elements appearing in the
+    `states` of each layers are suffixed with a '_<n>' before combining
+    them. The '<n>' is the number of the layer starting from 0.
+    The same process is true for the `contexts` and `outputs` of the
+    stack.
+
+    Parameters
+    ----------
+    transitions : list
+        List of recurrent units to use in each layer.
+        Note: A suffix with layer number is added to transitions' names.
+    prototype : :class:`.FeedForward`, optional
```

Comment:
Can you please rename it to `fork_prototype`? Simply `prototype` is a bit confusing.